project_euler/problem_54/test_poker_hand.py

```diff
@@ -0,0 +1,223 @@
+    ("2S 3H 6H 7S 9C", "7H 3C TH 6H 9S", "Loss"),
+    ("4S 5H 6H TS AC", "3S 5H 6H TS AC", "Win"),
+    ("2S AH 4H 5S 6C", "AD 4C 5H 6H 2C", "Tie"),
+    ("AS AH 3H AD AC", "AS AH 2H AD AC", "Win"),
+    ("AH AC 5H 5C QS", "AH AC 5H 5C KS", "Loss"),
+    ("AH AC 5H 5C QS", "KH KC 5H 5C QS", "Win"),
+    ("7C 7S KH 2H 7H", "3C 3S AH 2H 3H", "Win"),
+    ("3C 3S AH 2H 3H", "7C 7S KH 2H 7H", "Loss"),
+    ("6H 5H 4H 3H 2H", "5H 4H 3H 2H AH", "Win"),
+    ("5H 4H 3H 2H AH", "5H 4H 3H 2H AH", "Tie"),
+    ("5H 4H 3H 2H AH", "6H 5H 4H 3H 2H", "Loss"),
+    ("AH AD KS KC AC", "AH KD KH AC KC", "Win"),
+    ("2H 4D 3C AS 5S", "2H 4D 3C 6S 5S", "Loss"),
+    ("2H 3S 3C 3H 2S", "3S 3C 2S 2H 2D", "Win"),
+    ("4D 6D 5D 2D JH", "3S 8S 3H TC KH", "Loss"),
+    ("4S 6C 8S 3S 7S", "AD KS 2D 7D 7C", "Loss"),
+    ("6S 4C 7H 8C 3H", "5H JC AH 9D 9C", "Loss"),
+    ("9D 9H JH TC QH", "3C 2S JS 5C 7H", "Win"),
+    ("2H TC 8S AD 9S", "4H TS 7H 2C 5C", "Win"),
+    ("9D 3S 2C 7S 7C", "JC TD 3C TC 9H", "Loss"),
+]
+
+TEST_FLUSH = [
+    ("2H 3H 4H 5H 6H", True),
+    ("AS AH 2H AD AC", False),
+    ("2H 3H 5H 6H 7H", True),
+    ("KS AS TS QS JS", True),
+    ("8H 9H QS JS TH", False),
+    ("AS 3S 4S 8S 2S", True),
+]
+
+TEST_STRAIGHT = [
+    ("2H 3H 4H 5H 6H", True),
+    ("AS AH 2H AD AC", False),
+    ("2H 3H 5H 6H 7H", False),
+    ("KS AS TS QS JS", True),
+    ("8H 9H QS JS TH", True),
+]
+
+TEST_FIVE_HIGH_STRAIGHT = [
+    ("2H 4D 3C AS 5S", True, [(14, "S"), (2, "H"), (3, "C"), (4, "D"), (5, "S")]),
+    ("2H 5D 3C AS 5S", False, [(2, "H"), (3, "C"), (5, "D"), (5, "S"), (14, "S")]),
+    ("JH QD KC AS TS", False, [(10, "S"), (11, "H"), (12, "D"), (13, "C"), (14, "S")]),
+    ("9D 3S 2C 7S 7C", False, [(2, "C"), (3, "S"), (7, "C"), (7, "S"), (9, "D")]),
+]
+
+TEST_KIND = [
+    ("JH AH TH KH QH", 0),
+    ("JH 9H TH KH QH", 0),
+    ("JC KH JS JD JH", 7),
+    ("KH KC 3S 3H 3D", 6),
+    ("8C 9C 5C 3C TC", 0),
+    ("JS QS 9H TS KH", 0),
+    ("7C 7S KH 2H 7H", 3),
+    ("3C KH 5D 5S KH", 2),
+    ("QH 8H KD JH 8S", 1),
+    ("2D 6D 9D TH 7D", 0),
+]
+
+TEST_TYPES = [
+    ("JH AH TH KH QH", 23),
+    ("JH 9H TH KH QH", 22),
+    ("JC KH JS JD JH", 21),
+    ("KH KC 3S 3H 3D", 20),
+    ("8C 9C 5C 3C TC", 19),
+    ("JS QS 9H TS KH", 18),
+    ("7C 7S KH 2H 7H", 17),
+    ("3C KH 5D 5S KH", 16),
+    ("QH 8H KD JH 8S", 15),
+    ("2D 6D 9D TH 7D", 14),
+]
+
+
+def generate_random_hands():
+    N_RANDOM_TESTS = 100
+    RANDOM_TESTS = []
+    for _ in range(N_RANDOM_TESTS):
+        play, oppo = randrange(len(SORTED_HANDS)), randrange(len(SORTED_HANDS))
+        expected = ["Loss", "Tie", "Win"][(play >= oppo) + (play > oppo)]
+        hand, other = SORTED_HANDS[play], SORTED_HANDS[oppo]
+        RANDOM_TESTS.append((hand, other, expected))
+    return RANDOM_TESTS
+
+
+@pytest.mark.parametrize("hand, expected", TEST_FLUSH)
+def test_hand_is_flush(hand, expected):
+    player = PokerHand(hand)
+    assert player._is_flush() == expected
```

Comment:
```suggestion
    assert PokerHand(hand)._is_flush() == expected
```
The player is not so important as the hand itself.

Reply:
I will make the change but I don't see any difference. Can you tell me why are you suggesting this change?